src/engine/source/cmds/src/cmdRun.cpp

```diff
@@ -0,0 +1,163 @@
+#include "cmds/cmdRun.hpp"
+
+#include <atomic>
+#include <csignal>
+#include <thread>
+#include <vector>
+
+#include <hlp/hlp.hpp>
+#include <kvdb/kvdbManager.hpp>
+#include <logging/logging.hpp>
+#include <rxbk/rxFactory.hpp>
+
+#include "base/utils/getExceptionStack.hpp"
+#include "builder.hpp"
+#include "catalog.hpp"
+#include "engineServer.hpp"
+#include "protocolHandler.hpp"
+#include "register.hpp"
+
+namespace
+{
+constexpr auto WAIT_DEQUEUE_TIMEOUT_USEC = 1 * 1000000;
+
+// variables for handling threads
+std::atomic<bool> gs_doRun = true;
+std::vector<std::thread> gs_threadList;
+
+void sigint_handler(const int signum)
+{
+    // Inform threads that they must exit
+    gs_doRun = false;
+
+    for (auto& t : gs_threadList)
+    {
+        t.join();
+    };
+
+    // TODO: this should not be necessary, but server threads are not terminating.
+    exit(0);
+}
+} // namespace
+
+namespace cmd
+{
+void run(const std::string& kvdbPath,
+         const std::string& endpoint,
+         const int queueSize,
+         const int threads,
+         const std::string& fileStorage,
+         const std::string& environment,
+         const int logLevel)
```

Comment:
use a structure to group configuration parameters

Reply:
I agree with this

Reply:
Yes, the thing is once we implement the API and handle runtime and configuration the cmds module will change in its majority. For now, I think it's better to keep it simple.